src/main/java/novamachina/exnihilosequentia/common/init/ExNihiloTiles.java

```diff
@@ -44,18 +50,80 @@
-                    .of(StoneBarrelTile::new, ExNihiloBlocks.BARREL_STONE.get(), ExNihiloBlocks.BARREL_CRIMSON.get(),
-                            ExNihiloBlocks.BARREL_WARPED.get()).build(null));
+                    .of(StoneBarrelTile::new, getStoneBarrels()).build(null));
     private static final ExNihiloLogger logger = new ExNihiloLogger(LogManager.getLogger());
 
     private ExNihiloTiles() {
     }
 
+    private static Block[] getWoodBarrels() {
+        List<Block> blocksList = new ArrayList<>();
+        ArrayList<Block> list = new ArrayList<>(ForgeRegistries.BLOCKS.getValues());
+        for (Block block : list) {
+            ResourceLocation name = block.getRegistryName();
+            assert name != null;
+            if (block instanceof WoodBarrelBlock) {
+                blocksList.add(block);
+            }
+        }
+        return blocksList.toArray(new Block[0]);
+    }
+
+    private static Block[] getStoneBarrels() {
+        List<Block> blocksList = new ArrayList<>();
+        ArrayList<Block> list = new ArrayList<>(ForgeRegistries.BLOCKS.getValues());
+        for (Block block : list) {
+            ResourceLocation name = block.getRegistryName();
+            assert name != null;
+            if (block instanceof StoneBarrelBlock || block instanceof NetherBarrelBlock) {
+                blocksList.add(block);
+            }
+        }
+        return blocksList.toArray(new Block[0]);
+    }
+
+    private static Block[] getWoodCrucibles() {
+        List<Block> blocksList = new ArrayList<>();
+        ArrayList<Block> list = new ArrayList<>(ForgeRegistries.BLOCKS.getValues());
+        for (Block block : list) {
+            ResourceLocation name = block.getRegistryName();
+            assert name != null;
+            if (block instanceof WoodCrucibleBlock) {
+                blocksList.add(block);
+            }
+        }
+        return blocksList.toArray(new Block[0]);
+    }
+
+    private static Block[] getFiredCrucibles() {
+        List<Block> blocksList = new ArrayList<>();
+        ArrayList<Block> list = new ArrayList<>(ForgeRegistries.BLOCKS.getValues());
+        for (Block block : list) {
+            ResourceLocation name = block.getRegistryName();
+            assert name != null;
+            if (block instanceof FiredCrucibleBlock || block instanceof NetherCrucibleBlock) {
+                blocksList.add(block);
+            }
+        }
+        return blocksList.toArray(new Block[0]);
+    }
+
+    private static Block[] getSieves() {
+        List<Block> blocksList = new ArrayList<>();
+        ArrayList<Block> list = new ArrayList<>(ForgeRegistries.BLOCKS.getValues());
+        for (Block block : list) {
+            ResourceLocation name = block.getRegistryName();
+            assert name != null;
+            if (block instanceof WoodSieveBlock) {
+                blocksList.add(block);
+            }
+        }
+        return blocksList.toArray(new Block[0]);
+    }
+
```

Comment:
Why loop through all the blocks when it would be simpler to just create a static array?

Reply:
it isnt the finest idea as my first solution was tags but they arent bound to loop through them, so i just decided to use instanceof the blocks i need to resolve :|